homeassistant/components/camera/__init__.py

```diff
@@ -704,7 +725,19 @@ async def async_internal_added_to_hass(self) -> None:
         # Avoid calling async_refresh_providers() in here because it
         # it will write state a second time since state is always
         # written when an entity is added to hass.
-        self._webrtc_providers = await self._async_get_supported_webrtc_providers()
+        self._webrtc_provider = await self._async_get_supported_webrtc_provider(
+            async_get_supported_provider
+        )
+        self._legacy_webrtc_provider = await self._async_get_supported_webrtc_provider(
+            async_get_supported_legacy_provider
+        )
+
+        self.supports_async_webrtc_offer = (
+            inspect.signature(self.async_handle_web_rtc_offer).parameters.get(
```

Comment:
This feels weird to pick different code paths on the result of inspection? Why don't we use different methods, or even have the ones that do native webrtc override the built-in method, so there is no difference ?